script

```diff
@@ -834,6 +858,50 @@ only_send_error_notifications="0"
     remove_old_files_cmd+=(-delete)
   }
 
+  # remove old files
+  remove_old_files () {
+
+    # assign arguments to local variables for readability.
+    local -n _find_cmd=$1
+    local _type="$2"
+
+    # create variable equal to number_of_days_to_keep_backups plus one to make sure that there are files younger than the cutoff date.
+    local _days_plus_one=$((number_of_days_to_keep_backups + 1))
+    local _days_to_mins=$((24*60))
+
+    if [[ -n $("${_find_cmd[@]}" -mmin -$((_days_plus_one*_days_to_mins))) ]]; then
+      local _deleted_files=$("${_find_cmd[@]}" -mmin +$((number_of_days_to_keep_backups*_days_to_mins)) -delete -print)
```

Comment:
Oh interesting.  Truth be told, I haven't done a lot of shell scripting so a lot of this syntax is new to me.  (heavy C#, C, C++, and JS background)

Reply:
I was thinking about this some more, an I'm not sure if assigning the arguments to local variables suffers from the same problem. I would assume so, but I haven't had much chance to look into it. Something to consider anyway

Reply:
Yeah, I don't think it actually matters in this case because we don't care about the exit status anyway.  But, something to consider for sure, and might be worth changing just to be more 'proper'.  I'm not even sure if declaring it as 'local' is actually useful to begin with.

Reply:
yeah. I try to future proof against myself as much as possible with things like this. never know when I'm going to try to use it and then forget it won't work

Reply:
This should be fixed